src/frontend/screens/Settings/CreateOrJoinProject/CreateProject/index.tsx

```diff
@@ -0,0 +1,141 @@
+import {defineMessages, useIntl} from 'react-intl';
+import {NativeNavigationComponent} from '../../../../sharedTypes';
+import {Keyboard, KeyboardAvoidingView, StyleSheet, View} from 'react-native';
+import {useForm} from 'react-hook-form';
+import {Text} from '../../../../sharedComponents/Text';
+import * as React from 'react';
+import {
+  TouchableWithoutFeedback,
+  TouchableOpacity,
+} from 'react-native-gesture-handler';
+import {Button} from '../../../../sharedComponents/Button';
+import {BLACK, LIGHT_GREY, MEDIUM_GREY, RED} from '../../../../lib/styles';
+import MaterialIcon from 'react-native-vector-icons/MaterialIcons';
+import {HookFormTextInput} from '../../../../sharedComponents/HookFormTextInput';
+import {useCreateProject} from '../../../../hooks/server/projects';
+
+const m = defineMessages({
+  title: {
+    id: 'screens.Settings.CreateOrJoinProject.CreateProject.title',
+    defaultMessage: 'Create a Project',
+  },
+  enterName: {
+    id: 'screens.Settings.CreateOrJoinProject.enterName',
+    defaultMessage: 'Enter a name for the Project',
+  },
+  createProjectButton: {
+    id: 'screens.Settings.CreateOrJoinProject.createProjectButton',
+    defaultMessage: 'Create Project',
+  },
+  advancedSettings: {
+    id: 'screens.Settings.CreateOrJoinProject.advancedSettings',
+    defaultMessage: 'Advanced Project Settings',
+  },
+  importConfig: {
+    id: 'screens.Settings.CreateOrJoinProject.importConfig',
+    defaultMessage: 'Import Config',
+  },
+});
+
+type ProjectFormType = {
+  projectName: string;
+};
+
+export const CreateProject: NativeNavigationComponent<'CreateProject'> = ({
+  navigation,
+}) => {
+  const {formatMessage: t} = useIntl();
+  const [advancedSettingOpen, setAdvancedSettingOpen] = React.useState(false);
+  const {mutate} = useCreateProject();
+
+  const {
+    control,
+    handleSubmit,
+    formState: {errors},
+    watch,
+  } = useForm<ProjectFormType>({defaultValues: {projectName: ''}});
+
+  function handleCreateProject(val: ProjectFormType) {
+    mutate(val.projectName);
+    navigation.navigate('ProjectCreated', {name: val.projectName});
```

Comment:
this shouldn't navigate unless the mutation succeeds. was able to update the mutation implementation to throw an error and the navigation occurred anyways (and the project wasn't created despite the UI saying it was).

not sure what the "idiomatic" react query way of updating this is, but seems like [`mutateAsync`](https://tanstack.com/query/latest/docs/react/guides/mutations#promises) might be what's needed here?

```ts
try {
  await mutateAsync(val.projectName)
  navigation.navigate('ProjectCreated', {name: val.projectName});
} catch (err) {
  // Failed to create project, surface this error somewhere in the UI
}
```

Reply:
`mutate` actually as a [callback that you can use directly](https://tkdodo.eu/blog/mastering-mutations-in-react-query#mutate-or-mutateasync)

Reply:
We need to figure out what we are doing with our errors....

Reply:
https://github.com/digidem/CoMapeo-mobile/pull/77/commits/a30305e93468aedff6c33f3192648d423dff529b